executor/src/witgen/machines/write_once_memory.rs

```diff
@@ -135,6 +143,12 @@ impl<'a, T: FieldElement> WriteOnceMemory<'a, T> {
             value_polys,
             key_to_index,
             data: BTreeMap::new(),
+            public_names: parts
+                .fixed_data
+                .analyzed
+                .public_declarations_in_source_order()
```

Comment:
I think this is fragile, because this would include publics of other machines. I opened #2600 as a suggested fix, it also contains the other suggestions to this file.